### PR TITLE
API: Store name outside attrs

### PR DESCRIPTION
**Pull request by TomAugspurger**
This aligns with xarray and h5py:
https://github.com/pandas-dev/pandas/pull/29062#issuecomment-545703586

### CODE REVIEW

File: pandas/core/series.py [hunk 1]
```diff
@@ -159,7 +159,8 @@ class Series(base.IndexOpsMixin, generic.NDFrame):
 
     _typ = "series"
 
-    _metadata: List[str] = []
+    _name: Hashable
```
Comment (jbrockmendel) [inline pandas/core/series.py]: Optional hashable?

**Comment by jreback** [comment]
sure x- @jbrockmendel comment.

**Comment by jorisvandenbossche** [comment]
@TomAugspurger maybe also add a note somewhere that `attrs` is experimental?

File: pandas/core/generic.py [hunk 2]
```diff
@@ -231,6 +231,10 @@ def _init_mgr(self, mgr, axes=None, dtype=None, copy=False):
     def attrs(self) -> Dict[Optional[Hashable], Any]:
         """
         Dictionary of global attributes on this object.
+
+        .. warning::
+
+           attrs is experimental and may change without warning.
```
Comment (jorisvandenbossche) [inline pandas/core/generic.py]: Thanks! Can you also add this in the reference docs (where you added it in https://github.com/pandas-dev/pandas/pull/29062/files)

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by achapkowski** [comment]
When will this feature be not experimental?  Is allows for persistence of data when pickling on accessors.  I would encourage to remove the experimental tag.

**Comment by TomAugspurger** [comment]
You can remove the experimental label if you want. The basics aren't go to change. However, there are two future changes that will affect things in attrs

1. `__finalize__` isn't called everywhere: https://github.com/pandas-dev/pandas/issues/31452
2. We might provide an API for merging `.attrs` of multiple dataframes in operations like `concat`. This might be doable in a backwards-compat way. I'm not sure yet.

**Pull request merged**